### PR TITLE
Ericbrehault/sc 6549/cli allow to create kb

### PR DESCRIPTION
**Pull request by ebrehault**

### CODE REVIEW

File: nuclia/sdk/auth.py [hunk 1]
```diff
@@ -184,33 +191,29 @@ def post_login(self):
...
+        resp = requests.request(
+            method,
             path,
-            headers={"Authorization": f"Bearer {self._config.token}"},
+            **kwargs,
         )
-        if resp.status_code == 200:
+        if resp.status_code == 204:
+            return None
+        elif resp.status_code >= 200 and resp.status_code < 300:
```
Comment (lferran) [inline nuclia/sdk/auth.py:208]: I'm not sure there is a use-case right now, but 302 redirects would be raising an exception.

File: nuclia/sdk/auth.py [hunk 2]
```diff
@@ -184,33 +191,29 @@ def post_login(self):
...
-            raise UserTokenExpired()
-
-    def get_user(self, path: str):
+    def _request(self, method: str, path: str, data: Optional[Any] = None):
         if not self._config.token:
             raise NeedUserToken()
-        resp = requests.get(
+        kwargs = {"headers": {"Authorization": f"Bearer {self._config.token}"}}
+        if data is not None:
+            non_null_values = {k: v for k, v in data.items() if v is not None}
```
Comment (lferran) [inline nuclia/sdk/auth.py:199]: Not sure if there is a use-case, but null/None values may be required at some point, for instance to reset a specific attribute to None.

File: nuclia/sdk/auth.py [hunk 3]
```diff
@@ -184,33 +191,29 @@ def post_login(self):
...
         )
-        if resp.status_code == 200:
+        if resp.status_code == 204:
+            return None
+        elif resp.status_code >= 200 and resp.status_code < 300:
             return resp.json()
         elif resp.status_code == 403:
             raise UserTokenExpired()
+        else:
+            raise Exception(resp.text)
```
Comment (lferran) [inline nuclia/sdk/auth.py:213]: NIT: would be nice to distinguish client errors (4xx) from server errors (5xx) with different exception types, so the user can retry 502 for instance

File: nuclia/sdk/kbs.py [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
-from nuclia.config import retrieve
+from nuclia.config import KnowledgeBox, retrieve
 from nuclia.data import get_auth
-from nuclia.decorators import accounts
+from nuclia.decorators import account, accounts
 from nuclia.sdk.auth import NucliaAuth
 
-ADD_KB = BASE + "/api/v1/account/{account}/kbs"
+KBS_ENDPOINT = BASE + "/api/v1/account/{account}/kbs"
+KB_ENDPOINT = BASE + "/api/v1/account/{account}/kb/{slug}"
```
Comment (lferran) [inline nuclia/sdk/kbs.py:10]: if you want to use the KB slug you need to use `/kb/s/{slug}`: https://github.com/nuclia/nucliadb/blob/main/nucliadb/nucliadb/reader/api/v1/knowledgebox.py#L80

Comment (ebrehault) [reply nuclia/sdk/kbs.py:10]: Here we are on the global API, not on NucliaDB API, the kbid does not exist.

File: nuclia/sdk/auth.py (hunk 3 above)
Comment (ebrehault) [reply nuclia/sdk/auth.py]: I have added the status in the exception

**Review by lferran — APPROVED** [review]

**Pull request merged**